### ISSUE TITLE
Too many module sources?

### ISSUE THREAD

**Issue by mjg4243** [yihui/knitr#1628]
I am trying to run ```{python} in RMarkdown.  Some 'import' s work inside the chunk but others don't.

For instance - import pandas as pd 
works but 
import matplotlib.pyplot as plt
does not: ImportError: No module named matplotlib.pyplot

However -- all the modules ARE in "/usr/lib/python3/dist-packages"

It appears that RMarkdown does not like looking at more than one souce file.

I have also found that running the same python code in Thonny, it works OK if I add the statement

sys.path.append("/usr/lib/python3/dist-packages")

right before the import statement above --BUT

RMD does not recognize this statement and insstead throws a long error message which I bet you can duplicate on your own.

Wondering if this can be fixed in ```{python} -- sure would be nice

**Comment by cderv** [comment]
Hello, 

what is your setup ? Did you follow the advice on [reticulate vignette](https://rstudio.github.io/reticulate/articles/r_markdown.html#python-versionhttps://rstudio.github.io/reticulate/articles/r_markdown.html#python-version) ? 

Depending on the knitr version you have and the RStudio version you have, the behavior can be different in Rmarkdown. From knitr 1.18, it use the reticulate engine. You can [change the setup](https://rstudio.github.io/reticulate/articles/r_markdown.html#engine-setup) or change the [python version used](https://rstudio.github.io/reticulate/articles/r_markdown.html#python-version). 

If it uses `reticulate`, the [python configuration vignette](https://rstudio.github.io/reticulate/articles/versions.html) too to understand which python version is use and how virtualenv are setup also. 

With all that, you should check if Rmarkdown use the correct python version where the module is. 

I have no error with this small example: 
````r
---
title: "test python module"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(
	echo = TRUE,
	warning = FALSE,
	warnings = FALSE
)
```

```{python}
import matplotlib.pyplot as plt
```
````
You can use reticulate `py_config` or `py_discover_config()` to see what [python version are used](https://rstudio.github.io/reticulate/articles/versions.html#configuration-info). 

Hope it helps.

**Comment by mjg4243** [comment]
Did manage to figure it out 
import matplotlib # problem with tk backendmatplotlib.use('TkAgg') # this fixes it!import matplotlib.pyplot as plt

Mike Gallagher 

    On Sunday, December 9, 2018, 11:35:07 AM EST, Yihui Xie <notifications@github.com> wrote:  
 
 
Closed #1628.
